src/Http/Shared/CookieHeaderParserShared.cs

```diff
@@ -192,15 +198,14 @@ internal static StringSegment GetCookieValue(StringSegment input, ref int offset
             offset++;
         }
 
-        while (offset < input.Length)
+        var delta = input.AsSpan(offset).IndexOfAnyExcept(CookieValueChar);
+        if (delta == -1)
```

Comment:
```suggestion
        if (delta < 0)
```
Produces a tad better machine code for the CPU which can then fuse the comparison and the jump.

Reply:
I recall such myself too, although I was not able to measure any difference with the mentioned benchmark. Regardless, I apply the change, thank you for pointing out.